tutorials/llm/llama-3/nemo2-sft-peft/nemo2-sft.ipynb

```diff
@@ -0,0 +1,632 @@
+    "```\n",
+    "After the preprocessing logic, the data examples are transformed into NeMo format, as below:\n",
+    "```\n",
+    "{'input': \"Extract all the movies from this passage and the year they were released out. Write each movie as a separate sentence\\n\\nThe genre has existed since the early years of silent cinema, when Georges Melies' A Trip to the Moon (1902) employed trick photography effects. The next major example (first in feature length in the genre) was the film Metropolis (1927). From the 1930s to the 1950s, the genre consisted mainly of low-budget B movies. After Stanley Kubrick's landmark 2001: A Space Odyssey (1968), the science fiction film genre was taken more seriously. In the late 1970s, big-budget science fiction films filled with special effects became popular with audiences after the success of Star Wars (1977) and paved the way for the blockbuster hits of subsequent decades.\", 'output': 'A Trip to the Moon was released in 1902. Metropolis came out in 1927. 2001: A Space Odyssey was released in 1968. Star Wars came out in 1977.', 'category': 'information_extraction'}\n",
+    "```\n",
+    "Each data example is saved as a json string as one line in the `train/validation/test.jsonl` file, under `data_root` directory you specified earlier."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3: Run SFT with NeMo 2.0 API \n",
+    "\n",
+    "The following python script utilizes NeMo 2.0 API to perform SFT. In this script we are configuring the following components for training. These components are similar between SFT and PEFT. SFT and PEFT both uses `llm.finetune` API. To switch from PEFT to SFT you just need to remove `peft` parameter.\n",
+    "\n",
+    "### Trainer\n",
+    "NeMo 2.0 Trainer works simiarly to Pytorch Lightning trainer. You can specify to use MegatronStrategy as your model parallel strategy to use NVIDIA's Megatron-LM framework and pass in configurations as below:\n",
```

Comment:
revise sentences

"The NeMo 2.0 Trainer works similarly to the PyTorch Lightning trainer. You can specify the use of MegatronStrategy as your model parallel strategy to leverage NVIDIA's Megatron-LM framework and pass in configurations as shown below:"